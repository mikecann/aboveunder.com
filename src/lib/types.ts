export interface IPost
{
    id: string,
    title: string,
    headerImage: string;
    tags: string[],
    category: string;
    summary: string,
    markdownFileName: string;
    dateCreated: string,
}

export interface IPrint
{
    id: string,
    image:string;
    thumb:string;
    title: string,
    featured: boolean,
    description: string,
    dateCreated: string,
    printOptions: IPrintOption[]
}

export interface IPrintOption
{
    id:string,
    name: string;
    sizes: IPrintOptionSize[]
}

export interface IPrintOptionSize
{
    id:string,
    widthInches: number,
    heightInches: number,
    priceAUD: number,
    weight: number;
}

export interface IDB
{
    posts: IPost[],
    products: IPrint[]
}