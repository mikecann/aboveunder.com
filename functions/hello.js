"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function handler(event, context, callback) {
    console.log(event);
    callback(null, {
        statusCode: 200,
        body: JSON.stringify({ msg: "Hello, World!!" })
    });
}
exports.handler = handler;
