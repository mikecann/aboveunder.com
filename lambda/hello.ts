import { LambdaCallback } from "./types";

export function handler(event:any, context:any, callback:LambdaCallback) {
    console.log(event)
    callback(null, {
      statusCode: 200,
      body: JSON.stringify({msg: "Hello, World!"})
    })
  }