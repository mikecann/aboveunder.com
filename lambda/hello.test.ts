import { handler } from "./hello";
import { LambdaCallResult, LambdaCallbackError, LambdaFunction, LambdaCallbackResult } from "./types";

async function callFunction(event:any, context:any, lambda: LambdaFunction) : Promise<LambdaCallbackResult>
{
    return new Promise<LambdaCallbackResult>(resolve => {
        lambda(event, context, (error?:LambdaCallbackError, result?:LambdaCallResult) => {
            resolve({ error, result })
        })
    });
}

test('something', async () => 
{
    const result = await callFunction({},{}, handler);
    expect(result).toMatchSnapshot();
});