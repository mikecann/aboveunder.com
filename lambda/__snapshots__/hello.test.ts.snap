// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`something 1`] = `
Object {
  "error": null,
  "result": Object {
    "body": "{\\"msg\\":\\"Hello, World!\\"}",
    "statusCode": 200,
  },
}
`;
